File: quarkus/src/main/java/blewy/models/Product.java
package blewy.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Product {
    public String product_id;
    public String name;
    public String category;
    public double price;
    public boolean out_of_stock;
    public int stock;

    public Product() {
    }

    public Product(String product_id,
                   String name,
                   String category,
                   double price,
                   boolean out_of_stock,
                   int stock) {
        this.product_id = product_id;
        this.name = name;
        this.category = category;
        this.price = price;
        this.out_of_stock = out_of_stock;
        this.stock = stock;
    }
}

File: chewy-api/product.py
import uuid

class Product:
    def __init__(self, name: str, category: str, price: float, stock: int):
        self.product_id = str(uuid.uuid4())
        self.name = name
        self.category = category
        self.price = price
        self.stock = stock
        self.out_of_stock = False

    def to_dict(self) -> dict:
        return {
            "product_id": self.product_id,
            "name": self.name,
            "category": self.category,
            "price": self.price,
            "stock": self.stock,
            "out_of_stock": self.out_of_stock
        }

    @classmethod
    def from_dict(cls, data) -> "Product":
        obj = cls(
            name=data["name"],
            category=data["category"],
            price=data["price"],
            stock=data["stock"]
        )
        obj.product_id = data["product_id"]
        obj.out_of_stock = data.get("out_of_stock", False)
        return obj